drivers/MbedCRC.h

```diff
@@ -137,20 +148,18 @@ class MbedCRC {
     int32_t compute(void *buffer, crc_data_size_t size, uint32_t *crc)
     {
         MBED_ASSERT(crc != NULL);
-        int32_t status;
-        if (0 != (status = compute_partial_start(crc))) {
-            *crc = 0;
-            return status;
-        }
-        if (0 != (status = compute_partial(buffer, size, crc))) {
-            *crc = 0;
-            return status;
+        int32_t status = 0;
+        lock();
+        if (0 == (status = compute_partial_start(crc))) {
+            if (0 == (status = compute_partial(buffer, size, crc))) {
+                status = compute_partial_stop(crc);
+            }
         }
-        if (0 != (status = compute_partial_stop(crc))) {
+        if (0 != status) {
             *crc = 0;
-            return status;
         }
-        return 0;
+        unlock();
```

Comment:
nit: The series of if statements was cleaner IMO. Not a block, but in the future this may be better written with a series of unlocks in each return statement, or a goto to a cleanup block if the cleanup is too expensive.

Generally gotos are considered harmful, but this is one of the places where a goto makes the intention clearer than a series of nested if statements.

Reply:
I can have them as if's as cleanup is not expensive. But will avoid goto, will prefer nested if here as it won;t be deeper than this.